internal/juju/secrets_test.go

```diff
@@ -0,0 +1,246 @@
+	).Return("", errBoom).AnyTimes()
+
+	client := s.getSecretsClient()
+	output, err := client.CreateSecret(&CreateSecretInput{
+		ModelName: s.testModelName,
+		Name:      "test-secret",
+		Value:     map[string]string{"key": "value"},
+		Info:      "test info",
+	})
+	s.Require().Error(err)
+
+	s.Assert().Equal(output, CreateSecretOutput{})
+	s.Assert().Equal(errBoom, err)
+}
+
+func (s *SecretSuite) TestReadSecret() {
+	ctlr := s.setupMocks(s.T())
+	defer ctlr.Finish()
+
+	secretId := "secret:9m4e2mr0ui3e8a215n4g"
+	secretURI, err := coresecrets.ParseURI(secretId)
+	s.Require().NoError(err)
+	secretName := "test-secret"
+	secretRevision := 1
+
+	value := base64.StdEncoding.EncodeToString([]byte("value"))
+	s.mockSecretClient.EXPECT().ListSecrets(
+		true, coresecrets.Filter{
+			URI:      secretURI,
+			Label:    &secretName,
+			Revision: &secretRevision,
+		},
+	).Return([]apisecrets.SecretDetails{
+		{
+			Metadata: coresecrets.SecretMetadata{
+				Version: 1,
+			},
+			Revisions: []coresecrets.SecretRevisionMetadata{
+				{
+					Revision: 1,
+				},
+			},
+			Value: coresecrets.NewSecretValue(map[string]string{"key": value}),
+			Error: "",
+		},
+	}, nil).AnyTimes()
+
+	client := s.getSecretsClient()
+	output, err := client.ReadSecret(&ReadSecretInput{
+		SecretId:  secretId,
+		ModelName: s.testModelName,
+		Name:      &secretName,
+		Revision:  &secretRevision,
+	})
+	s.Require().NoError(err)
+
+	s.Assert().NotNil(output)
+	s.Assert().Equal(map[string]string{"key": "value"}, output.Value)
+}
+
+func (s *SecretSuite) TestReadSecretError() {
+	ctlr := s.setupMocks(s.T())
+	defer ctlr.Finish()
+
+	secretId := "secret:9m4e2mr0ui3e8a215n4g"
+	secretURI, err := coresecrets.ParseURI(secretId)
+	s.Require().NoError(err)
+
+	errBoom := errors.New("boom")
+	s.mockSecretClient.EXPECT().ListSecrets(
+		true, coresecrets.Filter{
+			URI: secretURI,
+		},
+	).Return([]apisecrets.SecretDetails{
+		{
+			Error: errBoom.Error(),
+		},
+	}, nil).AnyTimes()
+
+	client := s.getSecretsClient()
+	output, err := client.ReadSecret(&ReadSecretInput{
+		SecretId:  secretId,
+		ModelName: s.testModelName,
+	})
+	s.Require().Error(err)
+
+	s.Assert().Equal(output, ReadSecretOutput{})
+	s.Assert().Equal(errBoom, err)
+}
+
+func (s *SecretSuite) TestUpdateSecret() {
+	ctlr := s.setupMocks(s.T())
+	defer ctlr.Finish()
+
+	newSecretName := "test-secret2"
+	secretId := "secret:9m4e2mr0ui3e8a215n4g"
+	secretInfo := "secret info"
+	autoPrune := true
+
+	// update secret with value2
+	value2 := base64.StdEncoding.EncodeToString([]byte("value2"))
+	secretURI, err := coresecrets.ParseURI(secretId)
+	s.Require().NoError(err)
+
+	s.mockSecretClient.EXPECT().UpdateSecret(
+		secretURI, "", &autoPrune, newSecretName, "secret info", map[string]string{"key": value2},
+	).Return(nil).AnyTimes()
+
+	client := s.getSecretsClient()
+	err = client.UpdateSecret(&UpdateSecretInput{
+		SecretId:  secretId,
+		ModelName: s.testModelName,
+		NewName:   &newSecretName,
```

Comment:
It would be good to have a test where not all these values were set.

Reply:
Added new tests. With more reasonable names.